Updates for Terraform policy docs (#2864)
hashicorp/web-unified-docs: content/terraform-policy/v0.1.x (beta)/docs/policy/reference/policy/provider-policy.mdx

## Configuration model

The `provider_policy` block supports the following configuration:

- `provider_policy` &nbsp "[`<provider_name>`](#provider-name)" &nbsp "[`<policy_label>`](#policy-label)" &nbsp block | repeatable
  - [`enforce`](#enforce) &nbsp block | repeatable
    - [`condition`](#condition) &nbsp boolean expression | required
    - [`error_message`](#error-message) &nbsp string | optional
    - [`info_message`](#info-message) &nbsp string | optional

>   - [`enforcement_level`](#enforcement-level) &nbsp string | optional

Reviewer: ```suggestion
    - [`error_message`](#error_message) &nbsp string | optional
    - [`info_message`](#info_message) &nbsp string | optional
  - [`enforcement_level`](#enforcement_level) &nbsp string | optional
```

  - [`filter`](#filter) &nbsp boolean expression | optional
  - [`locals`](#locals) &nbsp block | optional

>     - `<name>` &nbsp any | repeatable

Reviewer: ```suggestion
    - [`<name>`](#locals) &nbsp any | repeatable
```

## Complete configuration example

> The following example demonstrates a provider_policy block with all configuration options specified:

Reviewer: ```suggestion
The following example demonstrates a `provider_policy` block with all configuration options specified:
```

}
```

</CodeBlockConfig>

## Specification

A `provider_policy` block supports the following configuration.

### Provider name

Specifies the provider name, such as `aws`, `google`, `azurerm`. You can use the `*` wildcard to match all providers.

### Policy label

Specifies a unique name for this policy within the policy file.

### `enforce`

> The `enforce` block defines a condition that must evaluate to `true` for the policy to pass. You can include one or more `enforce` blocks in each `provider_policy` block. If any of the enforce blocks' conditions fail during evaluation, then Terraform policy will fail the policy and return the corresponding error messages.

Reviewer: ```suggestion
The `enforce` block defines a condition that must evaluate to `true` for the policy to pass. You can include one or more `enforce` blocks in each `provider_policy` block. If any of the enforce blocks' conditions fail during evaluation, then Terraform policy fails the policy and returns the corresponding error messages.
```

- Data type: Block
- Repeatable

An `enforce` block supports the following configuration.

#### `condition`

A boolean expression that must evaluate to `true` for the policy to pass.

- Data type: Boolean expression
- Required

#### `error_message`

The error message to display when the condition evaluates to `false`.

- Data type: String
- Default: None

#### `info_message`

An informational message to display regardless of whether the policy passes or fails.

- Data type: String
- Default: None

### `enforcement_level`

Specifies the enforcement level for the policy. You can specify one of the following values:

> - `"advisory"`: Allows the operation to continue but prints a warning when if the conditions of the policy aren't met.

Reviewer 2: Typo: "when if the conditions" -- drop "if".
